plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -533,7 +582,18 @@ public void createTable(ConnectorSession session, ConnectorTableMetadata tableMe
     public JdbcOutputTableHandle beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata)
     {
         try {
-            return createTable(session, tableMetadata, generateTemporaryTableName());
+            if (supportsFaultTolerantExecution(session)) {
+                // Create the target table
+                createTable(session, tableMetadata);
+                // Create the temporary table
+                ColumnMetadata pageSinkIdColumn = getPageSinkIdColumn(
+                        tableMetadata.getColumns().stream().map(ColumnMetadata::getName).toList());
+                return createTable(session, tableMetadata, generateTemporaryTableName(), List.of(pageSinkIdColumn))
```

Comment:
as you are passing `pageSinkIdColumn` to `createTable` it should already register the `pageSInkIdColumnName` in returned `JdbcOutputTableHandle`. I suggest you change `createTable` to take `Optional<ColumnMetadata> pageSinkIdColumn` instead of `List<ColumnMetadata> extraColumns` and move the call to `.withPageSinkIdColumnName` to be part of `createTable`